internal/common/customplanmodifier/plan_modify_differ.go

```diff
@@ -0,0 +1,157 @@
+	statePlanDiff    []tftypes.ValueDiff
+}
+
+type UnknownInfo struct {
+	StateValue    attr.Value
+	UnknownValue  attr.Value
+	AttributeName string
+	StrPath       string
+	Path          path.Path
+}
+
+func (d *PlanModifyDiffer) Unknowns(ctx context.Context, diags *diag.Diagnostics) []UnknownInfo {
+	unknowns := []UnknownInfo{}
+	schema := d.schema
+	for _, diff := range d.statePlanDiff {
+		stateValue, tpfPath := conversion.AttributePathValue(ctx, diags, diff.Path, d.state, schema)
+		strPath := tpfPath.String()
+		planValue, _ := conversion.AttributePathValue(ctx, diags, diff.Path, d.plan, schema)
+		if planValue == nil || !planValue.IsUnknown() {
+			continue
+		}
+		unknowns = append(unknowns, UnknownInfo{
+			Path:          tpfPath,
+			StrPath:       strPath,
+			StateValue:    stateValue,
+			UnknownValue:  planValue,
+			AttributeName: conversion.AttributeName(tpfPath),
+		})
+	}
+	slices.SortFunc(unknowns, func(i, j UnknownInfo) int {
+		return strings.Compare(i.StrPath, j.StrPath)
+	})
+	return unknowns
+}
+
+func ReadPlanStructValue[T any](ctx context.Context, d *PlanModifyDiffer, p path.Path) *T {
+	return readSrcStructValue[T](ctx, d.plan, p)
+}
+
+func ReadStateStructValue[T any](ctx context.Context, d *PlanModifyDiffer, p path.Path) *T {
+	return readSrcStructValue[T](ctx, d.state, p)
+}
+
+func readSrcStructValue[T any](ctx context.Context, src conversion.TPFSrc, p path.Path) *T {
+	var obj types.Object
+	if localDiags := src.GetAttribute(ctx, p, &obj); localDiags.HasError() {
+		return nil
+	}
+	if obj.IsNull() || obj.IsUnknown() {
+		return nil
+	}
+	return conversion.TFModelObject[T](ctx, obj)
+}
+func ReadPlanStructValues[T any](ctx context.Context, d *PlanModifyDiffer, p path.Path, diags *diag.Diagnostics) []T {
+	return readSrcStructValues[T](ctx, d.plan, p, diags)
+}
+
+func readSrcStructValues[T any](ctx context.Context, src conversion.TPFSrc, p path.Path, diags *diag.Diagnostics) []T {
+	var objList types.List
+	if localDiags := src.GetAttribute(ctx, p, &objList); len(localDiags) > 0 {
+		diags.Append(localDiags...)
+		return nil
+	}
+	return conversion.TFModelList[T](ctx, diags, objList)
+}
+
+func UpdatePlanValue(ctx context.Context, diags *diag.Diagnostics, d *PlanModifyDiffer, p path.Path, value attr.Value) {
+	diags.Append(d.plan.SetAttribute(ctx, p, value)...)
+}
+
+func findChanges(ctx context.Context, diff []tftypes.ValueDiff, diags *diag.Diagnostics, schema conversion.TPFSchema) AttributeChanges {
+	changes := make(map[string]bool)
+	addChangeAndAncestorChanges := func(change path.Path) {
+		changes[change.String()] = true
+		for _, p := range conversion.AncestorPaths(change) {
+			changes[p.String()] = true
+		}
+	}
+	// avoids adding change for removed region_configs inside a removed replication_specs
+	isAncestorRemoved := func(p path.Path) bool {
+		for _, a := range conversion.AncestorPaths(p) {
+			if conversion.IsListIndex(a) {
+				if _, found := changes[conversion.AsRemovedIndex(a)]; found {
```

Comment:
```suggestion
				if changes[conversion.AsRemovedIndex(a)]{
```
should return zero value when not found

Reply:
Thanks. Added in 8d87c2375af0584f9c4765e51d1765ee498c1736